src/glmfit.jl

```diff
@@ -312,8 +313,8 @@ function fit(::Type{M},
     end
 
     rr = GlmResp(y, d, l, offset, wts)
-    res = M(rr, cholpred(X), false)
-    return dofit ? fit!(res; fitargs...) : res
+    res = M(rr, cholpred(X, allowrankdeficient,), false)
+    return dofit ? fit!(res; allowrankdeficient = allowrankdeficient, fitargs...) : res
```

Comment:
It's not... but I can make `allowrankdeficient` a field in the model, if you want?

The reason that I made it an argument here is because that is how it currently works when setting `allowrankdeficient=true` for linear models (it's an argument, not a field in the model), and I wanted to be consistent.

Reply:
@nalimilan, care to weight in on this? GLM will probably get some needed love after Terms 2.0. We could keep it like how it is now and open an issue for cleaning it in a future PR.

Reply:
I'm happy to make `allowrankdeficient` a field in this PR, or to do it in a future PR. Let me know what you prefer @Nosferican @nalimilan

Reply:
I don't have an opinion on this. Do you have a use case where it would be useful to have this information stored in the model?

Reply:
It would be good for packages such as ANOVA that build on it.